pkg/apiserver/internal/v1/instance.go

```diff
@@ -0,0 +1,250 @@
+/*
+Copyright 2020 The KubeCarrier Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/golang/protobuf/ptypes/empty"
+
+	v1 "github.com/kubermatic/kubecarrier/pkg/apiserver/api/v1"
+)
+
+type instanceServer struct {
+	client client.Client
+	mapper meta.RESTMapper
+}
+
+var _ v1.InstancesServer = (*instanceServer)(nil)
+
+func NewInstancesServer(c client.Client, mapper meta.RESTMapper) v1.InstancesServer {
+	return &instanceServer{
+		client: c,
+		mapper: mapper,
+	}
+}
+func (o instanceServer) Create(ctx context.Context, req *v1.InstanceCreateRequest) (res *v1.Instance, err error) {
+	obj := &unstructured.Unstructured{}
+
+	gvk, err := o.gvkFromInstance(o.mapper, req.Offering, req.Version)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "creating instance: unable to get Kind: %s", err.Error())
```

Comment:
This isn't internal error, but argument error.

Reply:
done